airflow/providers/airbyte/hooks/airbyte.py

```diff
@@ -0,0 +1,99 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import time
+from typing import Any, Optional
+
+from airflow.exceptions import AirflowException
+from airflow.providers.http.hooks.http import HttpHook
+
+
+class AirbyteHook(HttpHook):
+    """
+    Hook for Airbyte API
+
+    :param airbyte_conn_id: Required. The name of the Airflow connection to get
+        connection information for Airbyte.
+    :type airbyte_conn_id: str
+    :param api_version: Required. Airbyte API version.
+    :type api_version: str
+    """
+
+    RUNNING = "running"
+    SUCCEEDED = "succeeded"
+    CANCELLED = "cancelled"
+    PENDING = "pending"
+    FAILED = "failed"
+    ERROR = "error"
+
+    def __init__(self, airbyte_conn_id: str = "airbyte_default", api_version: str = "v1") -> None:
+        super().__init__(http_conn_id=airbyte_conn_id)
+        self.api_version: str = api_version
+
+    def wait_for_job(self, job_id: str, wait_seconds: int = 3, timeout: Optional[float] = None) -> None:
+        """
+        Helper method which polls a job to check if it finishes.
+
+        :param job_id: Id of the Airbyte job
+        :type job_id: str
+        :param wait_seconds: Number of seconds between checks
+        :type wait_seconds: int
+        :param timeout: How many seconds wait for job to be ready. Used only if ``asynchronous`` is False
+        :type timeout: float
+        """
+        state = None
+        start = time.monotonic()
+        while state not in (self.ERROR, self.SUCCEEDED, self.CANCELLED):
+            if timeout and start + timeout < time.monotonic():
+                raise AirflowException(f"Timeout: Airbyte job {job_id} is not ready after {timeout}s")
+            time.sleep(wait_seconds)
+            try:
+                job = self.get_job(job_id=job_id)
+                state = job.json()["job"]["status"]
+            except AirflowException as err:
+                self.log.info("Retrying. Airbyte API returned server error when waiting for job: %s", err)
```

Comment:
one thing that sticks out about this is, how do you know it's an error that should be a retry?

for example what if the requested job_id doesn't exist?

maybe that's just what you have to accept when using the httphook?

there is actually a `run_with_advanced_retry` which might be better and give you finer grain if desired.  but i don't think that needs to block this from being merged if you want to leave it 

perhaps @turbaszek has a thought here?

Reply:
yeah i mean looking at it more closely...

one thing is i'd put `continue` after logging retry.  for one this would signal more strongly what it's doing.  but additionally you don't need to proceed within this loop in a retry scenario --- you  need to try again.

but also, if the service is down or inaccessible do you want to keep trying for the full timeout duration? maybe better to limit retries to a given number with a stated retry interval that might not equal the job poke loop interval

Reply:
The `get_job` function calls the run then check_response from HttpHook. This function already manages responses different from 2XX or 3XX. Maybe this try/exception could be removed.
https://github.com/apache/airflow/blob/9f37af25ae7eb85fa8dbb70b7dbb23bbd5505323/airflow/providers/http/hooks/http.py#L143-L156

Reply:
if you want to leave it i'm good with it but i would suggest you add `continue` after retry scenario is encountered because there's no reason to finish the loop at that point

Reply:
I kept the exception and add a `continue`

Reply:
sorry the continue was actually a bad suggestion.  in my mind this bit was inside the loop
```python

        if state == self.ERROR:
            raise AirflowException(f"Job failed:\n{job}")
        if state == self.CANCELLED:
            raise AirflowException(f"Job was cancelled:\n{job}")
```
but my view of the indents was obscured by all the comments.

there's no point in using continue when it's already the end of the loop...